api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java

```diff
@@ -1,236 +1,141 @@
-
-    private Currency(
-            @Nullable UUID currencyId,
-            @Nullable Character currencyChar,
-            int roundedDigits,
-            double conversionCoefficient,
-            @Nullable Function<UUID, Double> startingBalance,
-            @NotNull BiFunction<Double, Locale, String> balanceFormatter,
-            @NotNull String @NotNull ... names
-    ) {
-        if (names.length < 1) {
-            throw new IllegalArgumentException("Empty array specified for currency name.");
-        }
-        this.balanceFormatter = Objects.requireNonNull(balanceFormatter, "balanceFormatter");
-        this.names = Objects.requireNonNull(names, "names");
-        this.currencyId = currencyId == null ? UUID.randomUUID() : currencyId;
-        this.currencyChar = currencyChar;
-        this.roundedDigits = roundedDigits;
-        this.conversionCoefficient = conversionCoefficient;
-        this.startingBalance = startingBalance == null ? ($) -> 0d : startingBalance;
-    }
+    /**
+     * Get's the currency's decimal character to be used for formatting purposes.
+     *
+     * @return The currency's decimal character.
+     * @author creatorfromhell
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    char decimal();
 
     /**
-     * Gets the {@link UUID} of the currency.
+     * Gets the currency's user-friendly display name.
      *
-     * @return the UUID of the currency
-     * @author lokka30, MrIvanPlays
+     * @return The currency's user-friendly display name.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    @NotNull
-    public UUID getCurrencyId() {
-        return currencyId;
-    }
+    String displayName();
 
     /**
-     * Gets the character which identifies this currency, which may be null.
+     * Gets the plural form of the currency's user-friendly display name.
      *
-     * @return char identifier. could be null
-     * @author MrIvanPlays
+     * @return The plural form of the currency's user-friendly display name.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    @Nullable
-    public Character getCurrencyCharacter() {
-        return currencyChar;
-    }
+    String displayNamePlural();
 
     /**
-     * Get the primary name of the currency, e.g. 'dollar' or 'token'.
+     * Gets the currency's default number of decimal digits when formatting this currency.
      *
-     * @return the name of the currency
-     * @author lokka30, MrIvanPlays
+     * @return The currency's default number of decimal digits when formatting this currency.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    @NotNull
-    public String getPrimaryCurrencyName() {
-        return names[0];
-    }
+    int precision();
```

Comment:
Should we prefix methods like `precision()` and `displayNamePlural()` with `get`?